chainforge/react-server/src/TabularDataNode.tsx

```diff
@@ -499,12 +499,22 @@ const TabularDataNode: React.FC<TabularDataNodeProps> = ({ data, id }) => {
 
   // Function to add multiple rows to the table
   const addMultipleRows = (newRows: TabularDataRowType[]) => {
+    // Remove the last row of the current table data as it is a blank row
+    const newTableData = tableData.slice(0, -1);
+
+    // Add the new rows to the table
     const addedRows = newRows.map((value) => {
-      const newRow: TabularDataRowType = { __uid: uuidv4(), ...value };
+      const newRow: TabularDataRowType = { __uid: uuidv4() };
+
+      // Map to correct column keys
+      tableColumns.forEach((col, index) => {
+        newRow[col.key] = value[`col-${index}`] || ""; // If undefined, default to empty string
+      });
+
       return newRow;
     });
 
-    setTableData((prev) => [...prev, ...addedRows]);
+    setTableData([...newTableData, ...addedRows]); // Update table data
```

Comment:
The reason it is like (prev) => is because of the way React updates, to avoid race conditions of state. I'm not sure if your version is free of race conditions. I would suggest since newTableData doesn't appear in the body of this function, to revert to a `(prev) =>` style update method. You can still do tableData.slice. By the way, you might need to double-check that tableData has rows to begin with (not sure how slice works on zero-length arrays).

Reply:
Added the change!